### PR TITLE
refactor: remove connection pool pooling

### PR DESCRIPTION
**Pull request by ernestas-vinted**
Removing connection pool pooling as it is ensured by the mcrouter.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -17,51 +11,8 @@ use crate::value::{FromMemcacheValueExt, ToMemcacheValue};
...
-    fn get_urls(self) -> Vec<String> {
-        let mut urls = vec![];
-        for url in self {
-            urls.push(url.to_string());
-        }
-        return urls;
-    }
-}
-
 #[derive(Clone)]
```
Comment (buinauskas) [inline src/client.rs]: Not possible to derive debug implementation?

File: src/client.rs [hunk 2]
```diff
@@ -71,81 +22,28 @@ pub(crate) fn check_key_len(key: &str) -> Result<(), MemcacheError> {
...
-        }
-        Ok(Client {
-            connections,
-            hash_function: default_hash_function,
-        })
-    }
+    /// Connect to memcache via given target Url
+    pub fn connect(target: impl AsRef<str>) -> Result<Self, MemcacheError> {
+        let pool = r2d2::Pool::builder()
+            .max_size(20)
```
Comment (buinauskas) [inline src/client.rs]: shouldn't size be an argument?

File: src/client.rs [hunk 3]
```diff
@@ -71,81 +22,28 @@ pub(crate) fn check_key_len(key: &str) -> Result<(), MemcacheError> {
...
-            connections: vec![pool],
-            hash_function: default_hash_function,
-        })
+        Ok(Self::with_pool(pool))
     }
 
-    pub fn connect<C: Connectable>(target: C) -> Result<Self, MemcacheError> {
-        Self::with_pool_size(target, 1)
+    /// Get pool connection
+    pub fn get_connection(&self) -> Result<PooledConnection<ConnectionManager>, r2d2::Error> {
```
Comment (buinauskas) [inline src/client.rs]: Do you own `MemcacheErrror`? I'd consider moving `r2d2::Error` to MemcacheError variants.

Comment (ernestas-vinted) [reply src/client.rs]: yep, good point will use MemcacheError

File: src/client.rs (hunk 1 above)
Comment (ernestas-vinted) [reply src/client.rs]: possible, derived :+1:

File: src/client.rs (hunk 2 above)
Comment (ernestas-vinted) [reply src/client.rs]: I was thinking about sane value and tradeoffs of exposing this as argument :thinking:, maybe compromise would be connnect_with_size ? on the other hand what about the connection_timeout :smile:

Comment (buinauskas) [reply src/client.rs]: Yeah, the rust way of doing things would be probably exposing another method such as `connnect_with_size` or creating a builder.

Comment (ernestas-vinted) [reply src/client.rs]: Left only wayt to initialize memcache `with_pool` function

**Pull request merged**